src/util.js

```diff
@@ -8,3 +10,9 @@ export function extend(obj, props) {
 }
 
 
+/** Coerces and returns a lowercased representation of the argument.
+ *  @param {mixed} str
+ */
+export function lowercase (str) {
```

Comment:
Originally I left everything calling `.toLowerCase()` on strings. However, in the diffing algorithm, it stringifies the node name and then passes that to a function because the function that gets it doesn't check if it's a string. I actually needed the non-stringified name for a function call after it.

I could have:

1. Used a different name
2. Passed the stringified value directly to the function without saving it as a variable

I noticed that several parts of the source was directly calling `toLowerCase()` on strings. If I could make this into a function that ensures it's a string, then I could remove the need for stringification just to pass of to the function and then be able to use the original value without refactoring much code around it in the differ.